plugin/kvictoria/kvictoria.go

```diff
@@ -0,0 +1,311 @@
+//	{namespace}_{subsystem}_group_manage_error{client_id="#{client_id}",node_id="#{node_id}"} (counter)
+//
+// Batches:
+//
+//	{namespace}_{subsystem}_produce_uncompressed_bytes_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//	{namespace}_{subsystem}_produce_compressed_bytes_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//	{namespace}_{subsystem}_produce_batches_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//	{namespace}_{subsystem}_produce_records_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//
+//	{namespace}_{subsystem}_fetch_uncompressed_bytes_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//	{namespace}_{subsystem}_fetch_compressed_bytes_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//	{namespace}_{subsystem}_fetch_batches_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//	{namespace}_{subsystem}_fetch_records_total{client_id="#{client_id}",node_id="#{node_id}",topic="#{topic}",partition="#{partition}"} (counter)
+//
+// # Usage
+//
+// This can be used in a client like this:
+//
+//	cl, err := kgo.NewCLient(
+//		kgo.WithHooks(kvictoria.NewMetrics("my_namespace")),
+//	)
+//
+// Note that you MUST use a new [Metrics] instance per client otherwise you can get surprising behaviour.
+//
+// Note that seed brokers use broker IDs prefixed with "seed_", with the number
+// corresponding to which seed it is.
+package kvictoria
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+	"net"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	vm "github.com/VictoriaMetrics/metrics"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+var (
+	// interface checks to ensure we implement the hooks properly
+	_ kgo.HookNewClient           = new(Metrics)
+	_ kgo.HookClientClosed        = new(Metrics)
+	_ kgo.HookBrokerConnect       = new(Metrics)
+	_ kgo.HookBrokerDisconnect    = new(Metrics)
+	_ kgo.HookBrokerWrite         = new(Metrics)
+	_ kgo.HookBrokerRead          = new(Metrics)
+	_ kgo.HookBrokerE2E           = new(Metrics)
+	_ kgo.HookBrokerThrottle      = new(Metrics)
+	_ kgo.HookGroupManageError    = new(Metrics)
+	_ kgo.HookProduceBatchWritten = new(Metrics)
+	_ kgo.HookFetchBatchRead      = new(Metrics)
+)
+
+// Metrics provides metrics using the [VictoriaMetrics/metrics] library.
+//
+// [VictoriaMetrics/metrics]: https://github.com/VictoriaMetrics/metrics
+type Metrics struct {
+	cfg cfg
+
+	clientID string
+	set      *vm.Set
+}
+
+// NewMetrics returns a new Metrics that tracks metrics under the given namespace.
```

Comment:
Definitely please document that a Metrics will panic if used for two kgo clients.